File: envConfig.js
function configMissing(configSetting) {
  throw new Error(`missing required .env setting: ${configSetting}`);
}

const envConfig = () => ({
  endpoint: process.env.ENDPOINT_CHK || configMissing('ENDPOINT_CHK'),
  testLocation: process.env.TEST_LOCATION || 'Test Location',
  updateOnChange: JSON.parse(process.env.UPDATE_ON_CHANGE ?? false),
  postmark: {
    token: process.env.POSTMARK_TOKEN || configMissing('POSTMARK_TOKEN'),
    emailto: process.env.EMAIL_TO || configMissing('EMAIL_TO'),
    emailfrom: process.env.EMAIL_FROM || configMissing('EMAIL_FROM'),
  },
  CloudFlare: {
    // new-style cf api key
    apiToken: process.env.CF_BEARER_TOKEN ?? null,
    // base api uri
    apiBase: 'https://api.cloudflare.com/client/v4/',
    // domain which has the dns record we're updating
    zoneDomain: process.env.CF_ZONE_DOMAIN ?? null,
    // the actualy dns record being updated (assuming type A);
    recordHostname: process.env.CF_DNS_RECORD_HOSTNAME ?? null,
  },
});

module.exports = envConfig();

File: python/envConfig.py
import os
import json
from dotenv import load_dotenv

def configMissing(configSetting):
  raise(ValueError("Missing required config setting: " + configSetting))

def load_config():
  load_dotenv()
  config = {
    'end_point': os.getenv("ENDPOINT_CHK") or configMissing("ENDPOINT_CHK"),
    'test_location': os.getenv("TEST_LOCATION") or 'Test Location',
    'update_on_change': json.loads(os.getenv('UPDATE_ON_CHANGE')) or False,
    'postmark': {
      'token': os.getenv("POSTMARK_TOKEN") or configMissing("POSTMARK_TOKEN"),
      'email_to': os.getenv("EMAIL_TO") or configMissing("EMAIL_TO"),
      'email_from': os.getenv("EMAIL_FROM") or configMissing("EMAIL_FROM"), 
    }
  }

  # print(f'have config: {config}')
  return config
  # if not os.getenv("POSTMARK_TO_EMAIL"):
  #   configMissing("POSTMARK_TO_EMAIL")
